orion/controllers/programs/programs_test.go

```diff
@@ -12,8 +12,9 @@ func TestExample(t *testing.T) {
 }
 
 func TestCheckProgram(t *testing.T) {
```

Comment:
I put this as a placeholder... but maybe we should separate this big mega test into 2 different tests. You just wrote code to check a Program's name and grades.
So let's have one unit test be all about checking program names `TestCheckProgramNames(...)`.
And another unit test be all about checking program grades `TestCheckProgramGrades(...)`.
In each unit test, you can then do a deep dive on checking all edges cases for the respective test.